protocol_connect.go

```diff
@@ -970,6 +979,8 @@ func (m *connectUnaryRequestMarshaler) buildGetURL(data []byte, compressed bool)
 }
 
 func (m *connectUnaryRequestMarshaler) writeWithGet(url *url.URL) *Error {
+	delete(m.header, connectHeaderProtocolVersion)
```

Comment:
This  seems like a good fix to retain, even though we don't want to convert timeouts to query params.